### ISSUE TITLE
Make CLI work again in Windows and macOS

### ISSUE THREAD

**Issue by micahflee** [micahflee/onionshare#357]
With the switch to PyInstaller again (#340) we lost the easy ability to have multiple binaries. So right now, the Windows and macOS builds are only including the GUI version.

This might not be feasibly with PyInstaller, but if it is, I'd like to bring back CLI.

**Comment by micahflee** [comment]
In any case, I'll need to make sure this wiki page stays updated:
https://github.com/micahflee/onionshare/wiki/Using-the-command-line-version

**Comment by micahflee** [comment]
This is looking less possible right now with PyInstaller, but I'll keep investigating: https://pyinstaller.readthedocs.io/en/stable/spec-files.html#multipackage-bundles

**Comment by micahflee** [comment]
Ok, I've figured out a hack that makes it so the macOS version can come with a CLI, but unfortunately it won't work for Windows.

Basically, instead of having PyInstaller execute `scripts/onionshare-gui` as the main script, it now executes `scripts/onionshare-pyinstaller`. This launcher script looks at the filename being launched and then either launches the CLI (or the filename is `onionshare`) or the GUI (if the filename is anything else, such as `onionshare-gui`).

PyInstaller now creates the binary `OnionShare.app/Contents/MacOS/onionshare-gui`, and the `build_osx.sh` script makes `onionshare` a symlink of `onionshare-gui`. So if you run either, you're technically running the same binary, but whether or not the GUI launches depends on which you execute.

This trick can't work for Windows because the CLI version there needs an actual change to the PyInstaller spec file, to set `console=True`. And the GUI version needs `console=False`, or else every time you open the GUI there will be a command prompt window open in the background. There's no way to conditionally set it to true when running `onionshare-gui.exe` because by the time this is executing, I've already used PyInstaller and set `console=False`.

So for now, CLI just won't be supported in Windows without building from source.
